tests/integration/nm/bond_test.py

```diff
@@ -78,9 +78,27 @@ def _bond_interface(name, options):
 
 
 @nmclient_context
-def _get_bond_current_state(name):
+def _get_bond_current_state(name, option=None):
+    """
+    When option defined, the returned state will only contains the
+    specified bond option and the bond mode.
+    When option not defined, the return state will only contains bond mode.
+    This is needed for assert check.
+    """
     nmdev = nm.device.get_device_by_name(name)
     nm_bond_info = nm.bond.get_bond_info(nmdev) if nmdev else {}
+    if not nm_bond_info:
+        return {}
+    bond_options = nm_bond_info[schema.Bond.OPTIONS_SUBTREE]
+    if not option:
+        nm_bond_info[schema.Bond.OPTIONS_SUBTREE] = {
+            schema.Bond.MODE: bond_options[schema.Bond.MODE]
+        }
+    else:
+        nm_bond_info[schema.Bond.OPTIONS_SUBTREE] = {
+            schema.Bond.MODE: bond_options[schema.Bond.MODE],
+            option: bond_options[option],
+        }
```

Comment:
IMO this way of filtering the data is not a good fit here.
If you just want to assert one or two specific values, change the assert in the test to check only for the relevant parameters or create a dedicated assert function.
Remember that the original asserts fitted well when we did not have support for reading actual properties.

But if you still prefer this way, how about trying to avoid duplicate code here:
```
nm_bond_info[schema.Bond.OPTIONS_SUBTREE] = {
    schema.Bond.MODE: bond_options[schema.Bond.MODE]
}
if option:
    nm_bond_info[option] = bond_options[option]
```